features/netsocket/nsapi_types.h

```diff
@@ -55,6 +55,14 @@ enum nsapi_error {
     NSAPI_ERROR_CONNECTION_TIMEOUT  = -3017,     /*!< connection timed out */
 };
 
+ typedef enum connection_status {
+    DISCONNECTED, 
```

Comment:
this should follow the rest of enumeration values (add prefix `NSAPI`), plus more specification (`NSAPI_CONNECTION_STATUS` or at least `NSAPI_STATUS`). See line 55 and others